.github/workflows/iroha2-dev.yml

```
@@ -12,84 +13,80 @@ env:
         with:
+          load: true
+          tags: iroha2:base
           context: .
-          push: true
-          tags: hyperledger/iroha2:dev
-          build-args: |
-            TARGET_DIR=release
-            PROFILE=--release
-      - name: Build and push docker image (iroha_client_cli)
-        uses: docker/build-push-action@v2
+          file: Dockerfile.base
+
+      - name: Build iroha2:build docker image
+        uses: docker/build-push-action@v3
         with:
+          load: true
+          tags: iroha2:build
           context: .
-          push: true
-          tags: hyperledger/iroha2:client-cli-dev
-          build-args: |
-            TARGET_DIR=release
-            PROFILE=--release
-            BIN=iroha_client_cli
-      - name: Build and push docker image (kagami)
-        uses: docker/build-push-action@v2
+          file: Dockerfile.build
+
+      - name: Build iroha2:dev docker image
+        run: |
+          docker build -f Dockerfile -t hyperledger/iroha2:dev .
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
         with:
-          context: .
-          push: true
-          tags: hyperledger/iroha2:kagami-dev
-          build-args: |
-            TARGET_DIR=release
-            PROFILE=--release
-            BIN=kagami
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Push iroha2:dev docker image
+        run: |
+          docker push hyperledger/iroha2:dev
 
       - name: Build and push docker image (load-rs:dev)
         run: |
           sleep 10s
           echo "wait to other workflow"
-      - uses: convictional/trigger-workflow-and-wait@v1.6.0
+      - uses: convictional/trigger-workflow-and-wait@v1.6.2
         with:
           owner: soramitsu
           repo: iroha2-longevity-load-rs
           github_token: ${{ secrets.G_ACCESS_TOKEN }}
           workflow_file_name: load-rs-push-from-dev.yaml
           ref: dev
 
-  build-ci-image:
+  upload_coverage:
+    if: github.event.pull_request.merged
     runs-on: ubuntu-latest
     container:
       image: 7272721/i2-ci:latest
     steps:
       - uses: actions/checkout@v2
-      - uses: Swatinem/rust-cache@v1
-      - name: Login to DockerHub
-        uses: docker/login-action@v1
+      - name: Download codecov report artifact
+        uses: dawidd6/action-download-artifact@v2.21.1
         with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Build & push Docker container
-        uses: docker/build-push-action@v2
+          workflow: iroha2-dev-pr.yml
+          name: lcov-${{ github.event.pull_request.head.sha }}.info
+          search_artifacts: true
+      - name: Upload to codecov
+        uses: codecov/codecov-action@v3
         with:
-          file: Dockerfile.ci
-          tags: hyperledger/iroha2-ci:latest
-          push: true
+          file: lcov.info
```

According to [the doc](https://github.com/codecov/codecov-action#arguments),

```suggestion
          files: lcov.info
```

Also, renaming from `lcov-${{ github.event.pull_request.head.sha }}.info` will be needed

---

Right, it is better to follow official arguments.

---

What renaming do you mean except [these ones](https://github.com/hyperledger/iroha/pull/2433/commits/e40617bd86203fa78cf38acd9d0a5da281ab9a76)?

---

As of 35a3dc6 in the `upload_coverage` job the download step downloads `lcov-${{ github.event.pull_request.head.sha }}.info` and the upload step tries to upload `lcov.info`

---

Actually codecov action is able to detect any possible reports to be uploaded:
```
[2022-07-19T09:09:15.167Z] ['info'] Searching for coverage files...
[2022-07-19T09:09:15.[26](https://github.com/BAStos525/soramitsu-iroha/runs/7405964287?check_suite_focus=true#step:5:27)4Z] ['info'] => Found 1 possible coverage files:
  lcov.info
```
That's why it was not an error. But right, it's better to specify the filename.